hwaddr.c

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+#include <stdio.h>
+#include <string.h>
+#include "hwaddr.h"
+
+/*
+ * Convert a MAC string (hwaddr) to bytes (data).
+ * hwaddr must be a null-terminated string in the format of xx:xx:xx:xx:xx:xx.
+ * data_buflen is typically 14, which is the sizeof sockaddr sa_data.
```

Comment:
always 14, no?

Reply:
Done -- hard-coded now to assume a 6-byte or more buffer.  Resolving conversation.